code/modules/martial_arts/sleeping_carp.dm

```diff
@@ -19,17 +20,30 @@
 	MARTIAL_ARTS_ACT_CHECK
 	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
 	var/atk_verb = pick("punches", "kicks", "chops", "hits", "slams")
-	D.visible_message("<span class='danger'>[A] [atk_verb] [D]!</span>", \
+	D.visible_message("<span class='danger'>[A] [atk_verb] [D]!</span>",
 					  "<span class='userdanger'>[A] [atk_verb] you!</span>")
 	D.apply_damage(rand(10,15), BRUTE)
-	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, 1, -1)
-	if(prob(50))
-		A.say(pick("HUAH!", "HYA!", "CHOO!", "WUO!", "KYA!", "HUH!", "HIYOH!", "CARP STRIKE!", "CARP BITE!"))
-	if(prob(D.getBruteLoss()) && !D.lying)
-		D.visible_message("<span class='warning'>[D] stumbles and falls!</span>", "<span class='userdanger'>The blow sends you to the ground!</span>")
-		D.Weaken(4)
+	playsound(get_turf(D), 'sound/weapons/punch1.ogg', 25, TRUE, -1)
 	add_attack_logs(A, D, "Melee attacked with martial-art [src] : Punched", ATKLOG_ALL)
 	return TRUE
 
 /datum/martial_art/the_sleeping_carp/explaination_header(user)
 	to_chat(usr, "<b><i>You retreat inward and recall the teachings of the Sleeping Carp...</i></b>")
+
+/datum/martial_art/the_sleeping_carp/teach(mob/living/carbon/human/H, make_temporary)
+	. = ..()
+	H.faction |= "carp"// :D
+	to_chat(H, "<span class='sciradio'>You have learned the ancient martial art of the Sleeping Carp! \
+					Your hand-to-hand combat has become much more effective, and you are now able to deflect any projectiles directed toward you when in throw mode. \
+					However, you are also unable to use any ranged weaponry. \
+					You can learn more about your newfound art by using the Recall Teachings verb in the Sleeping Carp tab.</span>")
+
+/datum/martial_art/the_sleeping_carp/remove(mob/living/carbon/human/H)
+	. = ..()
+	H.faction -= "carp"// :C
+
+/datum/martial_art/the_sleeping_carp/explaination_footer(user)
+	to_chat(user, "<b><i>In addition, by having your throw mode on when being shot at, you enter an active defensive mode where you will block and deflect all projectiles fired at you!</i></b>")
+
+/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
+	return ..() && user.in_throw_mode // in case an admin wants to var edit carp to have less deflection chance
```

Comment:
For the sake of avoiding a couple of proc calls:
```suggestion
/datum/martial_art/the_sleeping_carp/try_deflect(mob/user)
	return user.in_throw_mode && ..() // in case an admin wants to var edit carp to have less deflection chance
```